src/Symfony/Component/Validator/Constraints/UrlValidator.php

```diff
@@ -72,7 +72,7 @@ public function validate($value, Constraint $constraint)
         if ($constraint->checkDNS) {
             $host = parse_url($value, PHP_URL_HOST);
 
-            if (!checkdnsrr($host, 'ANY')) {
+            if ('' === $host || !checkdnsrr($host, 'ANY')) {
```

Comment:
Perhaps switch to `!is_string($host)` due possible `false` or `null` return values.